kentikapi/internal/api_payloads/saved_filter.go

```diff
@@ -0,0 +1,194 @@
+package api_payloads
+
+import (
+	"time"
+
+	"github.com/kentik/community_sdk_golang/kentikapi/internal/utils"
+	"github.com/kentik/community_sdk_golang/kentikapi/models"
+)
+
+type GetAllSavedFilterResponse []GetSavedFilterResponse
+
+func (p GetAllSavedFilterResponse) ToSavedFilters() ([]models.SavedFilter, error) {
+	var result []models.SavedFilter
+	err := utils.ConvertList(p, GetSavedFilterResponse.ToSavedFilter, &result)
+	if err != nil {
+		return nil, err
+	}
+	return result, err
+}
+
+type GetSavedFilterResponse struct {
+	savedFilterPayload
+}
+
+type CreateSavedFilterRequest struct {
+	savedFilterPayload
+}
+
+type CreateSavedFilterResponse struct {
+	savedFilterPayload
+}
+
+type UpdateSavedFilterRequest struct {
+	savedFilterPayload
+}
+
+type UpdateSavedFilterResponse struct {
+	savedFilterPayload
+}
+
+type savedFilterPayload struct {
+	ID                IntAsString    `json:"id,omitempty" response:"get,post,put"`
+	CompanyID         models.ID      `json:"company_id,string,omitempty" response:"get,post,put"`
```

Comment:
Please take a look: https://github.com/kentik/community_sdk_golang/pull/16/files#r595079608

Reply:
I see that the link to comment doesn't work somehow. I linked following comment:

> I see that ID and Company ID still have validation tags:
> 
> ```
> ID                IntAsString    `json:"id,omitempty" response:"get,post,put"`
> CompanyID         models.ID      `json:"company_id,string,omitempty" response:"get,post,put"`
> ```
> 
> Aren't they redundant?

Reply:
The validation is only for pointer, interface or slice (newly added):

validate.go:55
`if (field.Kind() == reflect.Ptr || field.Kind() == reflect.Interface || field.Kind() == reflect.Slice) && field.IsNil()`

So it does nothing for IntAsString and models.ID.

What's the purpose? I don't fully know. I used response tags to avoid nil-pointer dereference when I used `time.Time` in model and `*time.Time` in payload.

SNMPSpeed is IntAsString so request validation tag does nothing and is IMO redundant. Maybe the intention was to return error also on zero value of a field (so `0` for int), but this is not how current implementation works.

So I propose to remove those tags and move on.

Reply:
Tags allow to validate if the request/response payload has all the required attributes set to not-nil, so the user gets better error message than "Invalid request" from the API or than segfault when dereferencing nil in response payload. So it is mainly for pointer/interface types. Rarely for slices as slice can be assigned nil and API doesnt like this in some cases

Reply:
Ok, then I will remove these tags from savedFilterPayload's ID, CompanyID and InterfacePayload SNMPSpeed

Reply:
Resolved